<html>


<script>


function resolvedPromise(finished){
    return new Promise((resolve, reject) => {
        setTimeout(()=>{
            if(finished){
                resolve({'message':'delayed success!'})
            }
            else{
                reject('Did not resolve')
            }
        }, 500)
    })
}



function rejectedPromise(finished){
    return new Promise((resolve, reject)=>{
        setTimeout(()=>{
           
            let msg = ''

            try {
                throw new Error({ 'error': 'delayed excelption!' })
            } catch (e) {
                msg = e.message
            }

            if(false){
                resolve('there was no error')
            }
            else{
                reject(msg)
            }

        }, 500)
    })
}


rejectedPromise(true).then((result) =>{
    console.log(result)
}, error =>{
    console.log(error)
})

resolvedPromise(true).then((result) => {
    console.log(result)
}, error => {
    console.log(error)
})


</script>


</html>